nuxeo-runtime/nuxeo-connect-standalone/src/test/java/org/nuxeo/launcher/connect/TestConnectBroker.java

```diff
@@ -258,6 +260,36 @@ public void testInstallPackageRequest() throws Exception {
         checkLogEvents(expectedLogs, logCaptureResult.getCaughtEvents());
     }
 
+    /**
+     * Downloading a remote package fails if such package does not exist in remote. An explicit error message is shown:
+     * <pre>
+     * Download failed (not found).
+     * </pre>
+     */
+    @Test
+    @LogCaptureFeature.FilterWith(PkgRequestLogFilter.class)
+    // NXP-21573
+    public void testDownloadUnknownPackage() throws Exception {
+        // Environment setup
+        Environment.getDefault().setProperty(Environment.DISTRIBUTION_NAME, "server");
+        Environment.getDefault().setProperty(Environment.DISTRIBUTION_VERSION, "8.3");
+        ConnectBroker connectBroker = new ConnectBroker(Environment.getDefault());
+        ((StandaloneCallbackHolder) NuxeoConnectClient.getCallBackHolder()).setTestMode(true);
```

Comment:
I tried to refactor the code: extract this part to `@Before` each test, but then 2 tests failed:

- `testUpgradePackageRequestWithRelax`
- `testUpgradePackageRequestWithRelaxAndSnapshot`

Since this is not related to my changes, and they're related to environment variables. I prefer not to change it.